### PR TITLE
SDIT-1108: 🔧 Ensure build args supplied and correct app info format

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Stage: base image
-ARG BUILD_NUMBER=1_0_0
```
Comment (petergphillips) [inline Dockerfile:2]: these were being ignored as outside the `FROM`

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-ARG GIT_REF=not-available
-ARG GIT_BRANCH=not-available
-
 FROM node:18.18-bullseye-slim as base
 
+ARG BUILD_NUMBER
```
Comment (petergphillips) [inline Dockerfile:4]: no idea why we would want defaults for these

File: server/middleware/setUpHealthChecks.ts [hunk 3]
```diff
@@ -23,12 +23,14 @@ export default function setUpHealthChecks(applicationInfo: ApplicationInfo): Rou
...
   router.get('/info', (req, res) => {
     res.json({
-      uptime: process.uptime(),
+      git: {
+        branch: applicationInfo.branchName,
+      },
       build: {
-        buildNumber: applicationInfo.buildNumber,
-        gitRef: applicationInfo.gitRef,
+        artifact: applicationInfo.applicationName,
```
Comment (petergphillips) [inline server/middleware/setUpHealthChecks.ts:30]: change format to match kotlin

File: Dockerfile [hunk 4]
```diff
@@ -1,50 +1,56 @@
...
+
+# Define env variables for runtime health / info
+ENV BUILD_NUMBER=${BUILD_NUMBER}
+ENV GIT_REF=${GIT_REF}
+ENV GIT_BRANCH=${GIT_BRANCH}
 
 RUN apt-get update && \
-    apt-get upgrade -y
+        apt-get upgrade -y && \
+        apt-get autoremove -y && \
```
Comment (petergphillips) [inline Dockerfile:30]: copied latest from typescript template

File: server/config.ts [hunk 5]
```diff
@@ -101,10 +102,10 @@ export default {
     frontendComponents: {
       url: get('FRONTEND_COMPONENTS_URL', 'http://localhost:8083', requiredInProduction),
       timeout: {
-        response: Number(get('FRONTEND_COMPONENTS_TIMEOUT_RESPONSE', 10000)),
-        deadline: Number(get('FRONTEND_COMPONENTS_TIMEOUT_DEADLINE', 10000)),
+        response: Number(get('FRONTEND_COMPONENTS_TIMEOUT_RESPONSE', 5000)),
+        deadline: Number(get('FRONTEND_COMPONENTS_TIMEOUT_DEADLINE', 5000)),
```
Comment (petergphillips) [inline server/config.ts:106]: defaults were too high for frontend components, should be quite quick

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**